src/tof/detector.py

```diff
@@ -20,9 +22,37 @@ class Detector(Component):
     """
 
     def __init__(self, distance: sc.Variable, name: str = "detector"):
-        self.distance = distance
+        self.distance = distance.to(dtype=float, copy=False)
         self.name = name
-        super().__init__()
 
     def __repr__(self) -> str:
         return f"Detector(name={self.name}, distance={self.distance:c})"
+
+    def as_dict(self):
+        return {'distance': self.distance, 'name': self.name}
+
+
+@dataclass(frozen=True)
+class ReadonlyDetector(Component):
```

Comment:
I don't mind the "readonly" much, but I wonder if there is a clearer name for users? Something like `DetectorResult`, or something implying "this is what the detector sees"?

Reply:
ChatGPT suggests `DetectorReading`, quite good?

Reply:
:+1: